ee_plugin/utils.py

```diff
@@ -277,18 +279,28 @@ def ee_image_to_geotiff(
 
     tiles = tile_extent(ee_image, extent, scale)
     logger.info(f"Generated {len(tiles)} tiles for export.")
+    # TODO: review threshold and messaging
+    if len(tiles) > 5:
+        logger.warning(
+            "Exporting large number of tiles. Consider reducing scale or extent."
+        )
     tile_paths = []
 
-    for idx, tile in enumerate(tiles):
-        out_path = os.path.join(out_dir, f"{base_name}_tile{idx}.tif")
-        logger.info(f"Downloading tile {idx + 1}/{len(tiles)} to {out_path}")
-        download_tile(ee_image, tile, scale, projection, out_path)
-        tile_paths.append(out_path)
-
-    logger.info(f"Merging {len(tile_paths)} tiles into {merge_output}")
-    merge_geotiffs_gdal(tile_paths, merge_output)
-    for tile in tile_paths:
-        os.remove(tile)
+    try:
+        with tempfile.TemporaryDirectory() as temp_dir:
+            for idx, tile in enumerate(tiles):
+                if feedback and feedback.isCanceled():
+                    logger.info("Export cancelled by user.")
+                    raise StopIteration("Export cancelled by user.")
+                out_path = os.path.join(temp_dir, f"{base_name}_tile{idx}.tif")
+                logger.info(f"Downloading tile {idx + 1}/{len(tiles)} to {out_path}")
+                download_tile(ee_image, tile, scale, projection, out_path)
+                tile_paths.append(out_path)
+
+            logger.info(f"Merging {len(tile_paths)} tiles into {merge_output}")
+            merge_geotiffs_gdal(tile_paths, merge_output)
+    except StopIteration:
+        pass
```

Comment:
We were experimenting with some funky flow-control via the `StopIteration` exception, but now that we're using the `TemporaryDiectory()` context manager, we can simply use `return` to stop operation.

```suggestion
    with tempfile.TemporaryDirectory() as temp_dir:
        for idx, tile in enumerate(tiles):
            if feedback and feedback.isCanceled():
                logger.info("Export cancelled by user.")
                return
            out_path = os.path.join(temp_dir, f"{base_name}_tile{idx}.tif")
            logger.info(f"Downloading tile {idx + 1}/{len(tiles)} to {out_path}")
            download_tile(ee_image, tile, scale, projection, out_path)
            tile_paths.append(out_path)

        logger.info(f"Merging {len(tile_paths)} tiles into {merge_output}")
        merge_geotiffs_gdal(tile_paths, merge_output)
```